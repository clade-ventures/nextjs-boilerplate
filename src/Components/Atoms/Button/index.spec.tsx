/**
 * @jest-environment jsdom
 */

import * as React from "react";
import "@testing-library/jest-dom";

import { ButtonExample } from ".";

import { render, fireEvent } from "@Test/utils";

describe("ButtonExample", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <ButtonExample testID="button-example">Test</ButtonExample>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render with children prop", () => {
    const { getByText } = render(
      <ButtonExample testID="button-example">hey</ButtonExample>
    );

    expect(getByText("hey")).toBeTruthy();
  });

  it("should increment number on click", () => {
    let number = 1;
    const { getByTestId } = render(
      <ButtonExample
        testID="button-example"
        onClick={() => {
          number += 1;
        }}
      >
        ButtonExample
      </ButtonExample>
    );

    fireEvent.click(getByTestId("button-example"), new MouseEvent("click"));

    expect(number).toBe(2);
  });

  it("should render gradient blue button by gradientColor property", () => {
    const { getByTestId } = render(
      <ButtonExample testID="button-example" gradientColor="blue">
        ButtonExample
      </ButtonExample>
    );

    expect(getByTestId("button-example")).toHaveStyle(
      "background: linear-gradient(45deg, #2196F3 30%, #21CBF3 90%)"
    );
  });

  it("should render gradient red button by gradientColor property", () => {
    const { getByTestId } = render(
      <ButtonExample testID="button-example" gradientColor="red">
        ButtonExample
      </ButtonExample>
    );

    expect(getByTestId("button-example")).toHaveStyle(
      "background: linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)"
    );
  });
});
