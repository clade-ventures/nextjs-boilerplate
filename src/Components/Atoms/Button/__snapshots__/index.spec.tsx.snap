// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonExample should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    data-testid="button-example"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Test
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
